src/object.d

```diff
@@ -1857,10 +1857,14 @@ class TypeInfo_Interface : TypeInfo
 
     override size_t getHash(scope const void* p) @trusted const
     {
+        if (!*cast(void**)p)
+        {
+            return hashOf(null);
+        }
         Interface* pi = **cast(Interface ***)*cast(void**)p;
         Object o = cast(Object)(*cast(void**)p - pi.offset);
         assert(o);
-        return o.toHash();
+        return hashOf(o.toHash());
```

Comment:
If https://github.com/dlang/druntime/pull/2246 is accepted you will be able to revert this to `return o.toHash();`